web/cobrands/fixmystreet/assets.js

```diff
@@ -16,12 +16,61 @@ var fixmystreet = fixmystreet || {};
     };
 })();
 
+OpenLayers.Layer.VectorAsset = OpenLayers.Class(OpenLayers.Layer.Vector, {
+    initialize: function(name, options) {
+        OpenLayers.Layer.Vector.prototype.initialize.apply(this, arguments);
+        // But also want to do it immediately in case it's hiding the form or something
+        $(fixmystreet).on('report_new:category_change', this.update_layer_visibility.bind(this));
+        $(window).on('load', this.update_layer_visibility.bind(this));
```

Comment:
Could you call this in the `init_asset_layer` function, keep that all together?

Reply:
Just the load bit, I was meaning as init_asset_layer is already being set up to run in the same event.